### PR TITLE
#164074815 Refactor tests to work with the database and add travis-ci database

### PR DESCRIPTION
**Pull request by avpaul**
#### What does this PR do?
- Refactor tests to work with the database
- Add travis ci posgtress database integration

#### How should this be manually tested?
- Run travis-ci build 
- If you ave git and nodejs/npm installed  
  ```
      git clone repo_url
      npm install
      npm test
  ``` 
#### What are the relevant pivotal tracker stories?
#164074815 #164022932

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,8 +27,11 @@
     "debug": "~2.6.9",
     "dotenv": "^6.2.0",
     "express": "~4.16.0",
+    "express-jwt": "^5.3.1",
     "joi": "^14.3.1",
+    "jsonwebtoken": "^8.4.0",
     "morgan": "~1.9.0",
+    "nodemailer": "^5.1.1",
```
Comment (alexmochu) [inline package.json:34]: What does this "nodemailer" do?

**Comment by alexmochu** [comment]
- Your travis is still failing
- You have a typo on your pull request title. 'tavis' instead of 'travis'. You should practice attention to detail.
- Also, this PR has so many things that are not test related yet the pull request is about refactoring tests.

File: package.json (hunk 1 above)
Comment (avpaul) [reply package.json:34]: It is used to send an email from a nodejs application(password rest link)

**Comment by avpaul** [comment]
- The typo was corrected, thanks for the note.
- For the Travis CI build i'm working on it. All my tests are running successfully locally but when i push them only one fails.
- The PR covers other things because i was late to raise it. The branches i was working on are on the last Git repository. So when i created this repository i moved the develop branch and the latest branch.
@alexmochu

**Pull request merged**